File: 1. Two Sum/Main.py
"""
Simple Python solution
"""
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        indices = defaultdict(list)
        for i,num in enumerate(nums):
            other = target - num
            if other in indices:
                return indices[other],i
            indices[num] = i
        return -1,-1

File: 1. Two Sum/main.cpp
// Solution By Vandit Jain (jainvandit99)
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> solution;
        unordered_map<int,int> m;
        for(int i=0;i<nums.size();i++){
            if(m.find(nums[i]) != m.end()){
                solution.push_back(m[nums[i]]);
                solution.push_back(i);
                return solution;
            }
            m[target-nums[i]] = i;
        }
        
        return solution;
    }
};